python/dries_json_reader.py

```diff
@@ -48,6 +48,75 @@ def cells_json(data):
     return cell_dict
 
 
+def cell_sets_json(data):
+    '''
+    >>> data = {
+    ...     "cell_1": {
+    ...         "locations": [1632.02, -1305.7],
+    ...         "factors": {
+    ...             "pleiden_clus": [3],
+    ...             "kmeans": [8]
+    ...         },
+    ...         "mappings": {
+    ...             "tsne": [-11.0776, 6.0311],
+    ...             "umap": [-6.858, 15.7691]
+    ...         }
+    ...     }
+    ... }
+    >>> cell_sets = cell_sets_json(data)
+    >>> list(cell_sets.keys())
+    ['version', 'datatype', 'tree']
+    >>> cell_sets['datatype']
+    'cell'
+    >>> len(cell_sets['tree'])
+    2
+    >>> sorted([ n['name'] for n in cell_sets['tree'] ])
+    ['Leiden Clustering', 'k-means Clustering']
```

Comment:
(If it ever feels like this is more that should be crammed in a doctest, feel free to set up a standalone.: I do like to keep tests close, but if you find yourself doing actual coding inside the doctest, it can get painful.)

Reply:
Sounds good!